feature docs for scheduler preemption for in-place pod resize (#56348)
kubernetes/website: content/en/docs/tasks/configure-pod-container/resize-container-resources.md

  This is usually brief but might take longer depending on the resource type and runtime behavior.
  Any errors during actuation are reported in the `message` field (along with `reason: Error`).

### How deferred resizes are retried and preempted

If the requested resize is _Deferred_, the kubelet will periodically re-attempt the resize,
for example when another pod is removed or scaled down.

{{< note >}}
{{< feature-state feature_gate_name="InPlacePodVerticalScalingSchedulerPreemption" >}}

> When the `InPlacePodVerticalScalingSchedulerPreemption` feature gate is enabled, `kube-scheduler` actively evaluates `Deferred` resize requests on their assigned nodes. If a node lacks sufficient capacity to fulfill a higher-priority Pod's resize request, the scheduler can preempt (evict) lower-priority pods on that node to free up the required CPU or memory capacity. For more details, see [Preemption for in-place Pod resize](/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption-for-in-place-pod-resize).

Reviewer: ```suggestion
When the `InPlacePodVerticalScalingSchedulerPreemption` feature gate is enabled, `kube-scheduler` watches for pods with `Deferred` resize status. If a node lacks sufficient capacity to fulfill a higher-priority Pod's resize request, the scheduler can preempt (evict) lower-priority pods on that node to free up the required CPU or memory capacity. For more details, see [Preemption for in-place Pod resize](/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption-for-in-place-pod-resize).
```

Author: done!
